power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -709,25 +722,30 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
         };
     }
 
-    static auto scenario_update_restore(MainModelImpl& model, ConstDataset const& update_data,
-                                        SequenceIdx const& scenario_sequence, SequenceIdx& scenario_sequence_cache,
-                                        bool is_independent, std::vector<CalculationInfo>& infos) noexcept {
-        bool const do_update_cache = !is_independent;
+    static auto scenario_update_restore(
+        MainModelImpl& model, ConstDataset const& update_data,
+        std::array<std::reference_wrapper<std::vector<Idx2D> const>, n_types> const& scenario_sequence,
```

Comment:
i agree that that would be nicer. however, the fact that the amount of components in a scenario may differ means that you cannot use pass-by-value semantics and instead need a reference to the original vector

Reply:
Got it. I didn't thik about that and just didn't realice that once the size of `std::span` is set, it can not be changed, hence my proposal wouldn't make sense for sparse component updates, right?